File: TypeScript/codewars/Typescript/Scaling-Squared-Strings.ts
export const scale = (s: string, k: number, n: number): string => {
	if (s.trim() === "") return "";

	return s
		.split("\n")
		.map((line) => repeatLetters(line, k))
		.flatMap((scaledLine) => Array(n).fill(scaledLine))
		.join("\n");
};

function repeatLetters(str: string, quantity: number): string {
	return [...str].map((char) => char.repeat(quantity)).join("");
}

File: Csharp/CodeWars/src/Completed/ScalingSquaredStrings.cs
namespace CodeWars.Completed;

public class ScalingSquaredStrings
{
    public static string Scale(string strng, int k, int n)
    {
        if (string.IsNullOrEmpty(strng) || k <= 0 || n <= 0)
            return string.Empty;

        var lines = strng.Split('\n');
        List<string> scaledLines = [];

        foreach (var line in lines)
        {
            var scaledLine = string.Concat(line.Select(c => new string(c, k)));
            for (int i = 0; i < n; i++)
                scaledLines.Add(scaledLine);
        }

        return string.Join('\n', scaledLines);
    }
}
